finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

```diff
@@ -143,9 +148,9 @@ object Trace {
     val spanId = SpanId(rng.nextLong())
     idOption match {
       case Some(id) =>
-        TraceId(Some(id.traceId), Some(id.spanId), spanId, id.sampled, id.flags)
+        TraceId(Some(id.traceId), Some(id.spanId), spanId, id.sampled, id.flags, id.traceIdHigh)
       case None =>
-        TraceId(None, None, spanId, None, Flags())
+        TraceId(None, None, spanId, None, Flags(), None)
```

Comment:
Also an aside,  but I'm surprised we don't have a `TraceId.Empty` instance laying about.